Cannot build with USE_REGEX=OFF

I was trying to build with `USE_REGEX=OFF` and there's an undefined reference:

```
atch.c: In function 'f_match_inv':
Match.c:462:2: error: 'warn_nonzero_min_sec' undeclared (first use in this function)
  462 |  warn_nonzero_min_sec = 0;
      |  ^~~~~~~~~~~~~~~~~~~~
Match.c:462:2: note: each undeclared identifier is reported only once for each function it appears in
Match.c: In function 'f_Match_inv':
Match.c:475:2: error: 'warn_nonzero_min_sec' undeclared (first use in this function)
  475 |  warn_nonzero_min_sec = 0;
      |  ^~~~~~~~~~~~~~~~~~~~
```

`warn_nonzero_min_sec` is defined as `extern`, but within an `#IFDEF USE_REGEX`

This applies to both the CMake build and wgrib2 3.0

--- a/wgrib2/Match.c
+++ b/wgrib2/Match.c
@@ -5,6 +5,8 @@
 #include "wgrib2.h"
 #include "fnlist.h"
 
+extern int warn_nonzero_min_sec;
+
 #ifdef USE_REGEX
 #include <sys/types.h>
 #include <regex.h>
@@ -41,7 +43,6 @@ int regex_type;
 static regex_t egrep_preg[GREP_MAX];
 static int egrep_type[GREP_MAX];
 
-extern int warn_nonzero_min_sec;
 /*
  *  match, if use extended regex library call (like in egrep)
  * 

